From: Jo-Philipp Wich <jow@openwrt.org>
Date: Sat, 21 Apr 2012 03:02:39 +0000
Subject: [PATCH] gcc: make getenv() spec function nonfatal

Make the getenv() spec function nonfatal if requested environment variable is unset.

Signed-off-by: Jo-Philipp Wich <jow@openwrt.org>
---
--- a/gcc/gcc.cc
+++ b/gcc/gcc.cc
@@ -10184,8 +10184,10 @@ getenv_spec_function (int argc, const ch
     }
 
   if (!value)
-    fatal_error (input_location,
-		 "environment variable %qs not defined", varname);
+    {
+      warning (input_location, "environment variable %qs not defined", varname);
+      value = "";
+    }
 
   /* We have to escape every character of the environment variable so
      they are not interpreted as active spec characters.  A
